### PR TITLE
[low-power] fix csl code wrapper

### PR DESCRIPTION
**Pull request by Irving-cl**
This PR updates the wrapper for CSL code, updating most occurrences `!OPENTHREAD_MTD` to `OPENTHREAD_FTD`.

As I remember, we use `!OPENTHREAD_MTD` because we think both FTD and RCP need that code. But that actually means host side of RCP (posix) need that code. And posix host is always FTD:
```
target_link_libraries(ot-daemon PRIVATE                                                                                                                                                                                                                                                                           
    openthread-cli-ftd                                                                                                                                                                                                                                                                                            
    ${OT_PLATFORM_LIB}                                                                                                                                                                                                                                                                                            
    openthread-ftd                                                                                                                                                                                                                                                                                                
    ${OT_PLATFORM_LIB}                                                                                                                                                                                                                                                                                            
    openthread-ncp-ftd                                                                                                                                                                                                                                                                                            
   ${OT_MBEDTLS}                                                                                                                                                                                                                                                                                                 
)
```
The radio part itself don't need those code. The radio part doesn't even contain most of those source files that has CSL logic (except for `sub_mac.cpp`):
```
target_sources(openthread-radio PRIVATE                                                                                                                                                                                                                                                                               
    api/diags_api.cpp                                                                                                                                                                                                                                                                                                 
    api/instance_api.cpp                                                                                                                                                                                                                                                                                              
    api/link_raw_api.cpp                                                                                                                                                                                                                                                                                              
    api/logging_api.cpp                                                                                                                                                                                                                                                                                               
    api/random_noncrypto_api.cpp                                                                                                                                                                                                                                                                                      
    api/tasklet_api.cpp                                                                                                                                                                                                                                                                                               
    common/instance.cpp                                                                                                                                                                                                                                                                                               
    common/logging.cpp                                                                                                                                                                                                                                                                                                
    common/random_manager.cpp                                                                                                                                                                                                                                                                                         
    common/string.cpp                                                                                                                                                                                                                                                                                                 
    common/tasklet.cpp                                                                                                                                                                                                                                                                                                
    common/timer.cpp                                                                                                                                                                                                                                                                                                  
    crypto/aes_ccm.cpp                                                                                                                                                                                                                                                                                                
    crypto/aes_ecb.cpp                                                                                                                                                                                                                                                                                                
    diags/factory_diags.cpp                                                                                                                                                                                                                                                                                           
    mac/link_raw.cpp                                                                                                                                                                                                                                                                                                  
    mac/mac_frame.cpp                                                                                                                                                                                                                                                                                                 
    mac/mac_types.cpp                                                                                                                                                                                                                                                                                                 
    mac/sub_mac.cpp                                                                                                                                                                                                                                                                                                   
    mac/sub_mac_callbacks.cpp                                                                                                                                                                                                                                                                                         
    radio/radio.cpp                                                                                                                                                                                                                                                                                                   
    radio/radio_callbacks.cpp                                                                                                                                                                                                                                                                                         
    radio/radio_platform.cpp                                                                                                                                                                                                                                                                                          
    thread/link_quality.cpp                                                                                                                                                                                                                                                                                           
    utils/lookup_table.cpp                                                                                                                                                                                                                                                                                            
    utils/parse_cmdline.cpp                                                                                                                                                                                                                                                                                           
)
```
So I updated all the occurrences of relevant `!OPENTHREAD_MTD` in all source files except for `sub_mac.cpp`.

This PR also adds missing initialization for data field in `Mac`.

### CODE REVIEW

**Review by brianljt — APPROVED** [review]
LGTM👍 Thanks!

File: src/core/mac/mac.cpp [hunk 1]
```diff
@@ -109,7 +112,7 @@ Mac::Mac(Instance &aInstance)
 #if OPENTHREAD_FTD
     , mMaxFrameRetriesIndirect(kDefaultMaxFrameRetriesIndirect)
 #endif
-#if !OPENTHREAD_MTD && OPENTHREAD_CONFIG_MAC_CSL_TRANSMITTER_ENABLE
+#if OPENTHREAD_FTD && OPENTHREAD_CONFIG_MAC_CSL_TRANSMITTER_ENABLE
```
Comment (jwhui) [inline src/core/mac/mac.cpp:115]: Combine with above `#if OPENTHREAD_FTD`?

File: src/core/mac/mac.cpp [hunk 2]
```diff
@@ -207,7 +210,7 @@ bool Mac::IsInTransmitState(void) const
     case kOperationTransmitDataDirect:
 #if OPENTHREAD_FTD
     case kOperationTransmitDataIndirect:
-#if !OPENTHREAD_MTD && OPENTHREAD_CONFIG_MAC_CSL_TRANSMITTER_ENABLE
+#if OPENTHREAD_FTD && OPENTHREAD_CONFIG_MAC_CSL_TRANSMITTER_ENABLE
```
Comment (jwhui) [inline src/core/mac/mac.cpp:213]: Combine with above `#if OPENTHREAD_FTD`?

File: src/core/mac/mac.cpp [hunk 3]
```diff
@@ -772,7 +775,7 @@ void Mac::PerformNextOperation(void)
         mPendingTransmitDataDirect = false;
 #if OPENTHREAD_FTD
         mPendingTransmitDataIndirect = false;
-#if !OPENTHREAD_MTD && OPENTHREAD_CONFIG_MAC_CSL_TRANSMITTER_ENABLE
+#if OPENTHREAD_FTD && OPENTHREAD_CONFIG_MAC_CSL_TRANSMITTER_ENABLE
```
Comment (jwhui) [inline src/core/mac/mac.cpp:778]: Combine with above `#if OPENTHREAD_FTD`?

File: src/core/mac/mac.cpp [hunk 4]
```diff
@@ -1423,7 +1426,7 @@ void Mac::RecordFrameTransmitStatus(const TxFrame &aFrame,
 #if OPENTHREAD_FTD
         if (aAckFrame->GetVersion() == Frame::kFcfFrameVersion2015)
         {
-#if !OPENTHREAD_MTD && OPENTHREAD_CONFIG_MAC_CSL_TRANSMITTER_ENABLE
+#if OPENTHREAD_FTD && OPENTHREAD_CONFIG_MAC_CSL_TRANSMITTER_ENABLE
```
Comment (jwhui) [inline src/core/mac/mac.cpp:1429]: Combine with above `#if OPENTHREAD_FTD`?

File: src/core/mac/mac.cpp [hunk 5]
```diff
@@ -1198,7 +1201,7 @@ void Mac::BeginTransmit(void)
         }
         break;
 
-#if !OPENTHREAD_MTD && OPENTHREAD_CONFIG_MAC_CSL_TRANSMITTER_ENABLE
+#if OPENTHREAD_FTD && OPENTHREAD_CONFIG_MAC_CSL_TRANSMITTER_ENABLE
```
Comment (jwhui) [inline src/core/mac/mac.cpp:1204]: Combine with above `#if OPENTHREAD_FTD`?

File: src/core/mac/mac.cpp [hunk 6]
```diff
@@ -870,7 +873,7 @@ void Mac::PerformNextOperation(void)
     case kOperationTransmitDataDirect:
 #if OPENTHREAD_FTD
     case kOperationTransmitDataIndirect:
-#if !OPENTHREAD_MTD && OPENTHREAD_CONFIG_MAC_CSL_TRANSMITTER_ENABLE
+#if OPENTHREAD_FTD && OPENTHREAD_CONFIG_MAC_CSL_TRANSMITTER_ENABLE
```
Comment (jwhui) [inline src/core/mac/mac.cpp:876]: Combine with above `#if OPENTHREAD_FTD`?

File: src/core/mac/mac.cpp [hunk 7]
```diff
@@ -727,7 +730,7 @@ void Mac::StartOperation(Operation aOperation)
         mPendingTransmitDataIndirect = true;
         break;
 
-#if !OPENTHREAD_MTD && OPENTHREAD_CONFIG_MAC_CSL_TRANSMITTER_ENABLE
+#if OPENTHREAD_FTD && OPENTHREAD_CONFIG_MAC_CSL_TRANSMITTER_ENABLE
```
Comment (jwhui) [inline src/core/mac/mac.cpp:733]: Combine with above `#if OPENTHREAD_FTD`?

File: src/core/mac/mac.cpp [hunk 8]
```diff
@@ -581,7 +584,7 @@ void Mac::RequestIndirectFrameTransmission(void)
     return;
 }
 
-#if !OPENTHREAD_MTD && OPENTHREAD_CONFIG_MAC_CSL_TRANSMITTER_ENABLE
+#if OPENTHREAD_FTD && OPENTHREAD_CONFIG_MAC_CSL_TRANSMITTER_ENABLE
```
Comment (jwhui) [inline src/core/mac/mac.cpp:587]: Combine with above `#if OPENTHREAD_FTD`?

File: src/core/common/instance.hpp [hunk 9]
```diff
@@ -544,7 +544,7 @@ template <> inline DataPollHandler &Instance::Get(void)
     return mThreadNetif.mMeshForwarder.mIndirectSender.mDataPollHandler;
 }
 
-#if !OPENTHREAD_MTD && OPENTHREAD_CONFIG_MAC_CSL_TRANSMITTER_ENABLE
+#if OPENTHREAD_FTD && OPENTHREAD_CONFIG_MAC_CSL_TRANSMITTER_ENABLE
```
Comment (jwhui) [inline src/core/common/instance.hpp:547]: Combine with above `#if OPENTHREAD_FTD`?

File: src/core/mac/mac.cpp [hunk 10]
```diff
@@ -1630,7 +1633,7 @@ void Mac::HandleTransmitDone(TxFrame &aFrame, RxFrame *aAckFrame, otError aError
         break;
 
 #if OPENTHREAD_FTD
-#if !OPENTHREAD_MTD && OPENTHREAD_CONFIG_MAC_CSL_TRANSMITTER_ENABLE
+#if OPENTHREAD_FTD && OPENTHREAD_CONFIG_MAC_CSL_TRANSMITTER_ENABLE
```
Comment (jwhui) [inline src/core/mac/mac.cpp:1636]: Combine with above `#if OPENTHREAD_FTD`?

File: src/core/mac/mac.hpp [hunk 11]
```diff
@@ -221,7 +221,7 @@ class Mac : public InstanceLocator, private NonCopyable
      */
     void RequestIndirectFrameTransmission(void);
 
-#if !OPENTHREAD_MTD && OPENTHREAD_CONFIG_MAC_CSL_TRANSMITTER_ENABLE
+#if OPENTHREAD_FTD && OPENTHREAD_CONFIG_MAC_CSL_TRANSMITTER_ENABLE
```
Comment (jwhui) [inline src/core/mac/mac.hpp:224]: Combine with above `#if OPENTHREAD_FTD`?

File: src/core/mac/mac.hpp [hunk 12]
```diff
@@ -792,7 +792,7 @@ class Mac : public InstanceLocator, private NonCopyable
         kOperationTransmitDataDirect,
 #if OPENTHREAD_FTD
         kOperationTransmitDataIndirect,
-#if !OPENTHREAD_MTD && OPENTHREAD_CONFIG_MAC_CSL_TRANSMITTER_ENABLE
+#if OPENTHREAD_FTD && OPENTHREAD_CONFIG_MAC_CSL_TRANSMITTER_ENABLE
```
Comment (jwhui) [inline src/core/mac/mac.hpp:795]: Combine with above `#if OPENTHREAD_FTD`?

File: src/core/mac/mac.hpp [hunk 13]
```diff
@@ -920,7 +920,7 @@ class Mac : public InstanceLocator, private NonCopyable
 #if OPENTHREAD_FTD
     uint8_t mMaxFrameRetriesIndirect;
 #endif
-#if !OPENTHREAD_MTD && OPENTHREAD_CONFIG_MAC_CSL_TRANSMITTER_ENABLE
+#if OPENTHREAD_FTD && OPENTHREAD_CONFIG_MAC_CSL_TRANSMITTER_ENABLE
```
Comment (jwhui) [inline src/core/mac/mac.hpp:923]: Combine with above `#if OPENTHREAD_FTD`?

File: src/core/thread/indirect_sender.cpp [hunk 14]
```diff
@@ -65,7 +65,7 @@ IndirectSender::IndirectSender(Instance &aInstance)
     , mEnabled(false)
     , mSourceMatchController(aInstance)
     , mDataPollHandler(aInstance)
-#if !OPENTHREAD_MTD && OPENTHREAD_CONFIG_MAC_CSL_TRANSMITTER_ENABLE
+#if OPENTHREAD_FTD && OPENTHREAD_CONFIG_MAC_CSL_TRANSMITTER_ENABLE
```
Comment (jwhui) [inline src/core/thread/indirect_sender.cpp:68]: Combine with above `#if OPENTHREAD_FTD`?

File: src/core/thread/indirect_sender.cpp [hunk 15]
```diff
@@ -82,7 +82,7 @@ void IndirectSender::Stop(void)
     }
 
     mDataPollHandler.Clear();
-#if !OPENTHREAD_MTD && OPENTHREAD_CONFIG_MAC_CSL_TRANSMITTER_ENABLE
+#if OPENTHREAD_FTD && OPENTHREAD_CONFIG_MAC_CSL_TRANSMITTER_ENABLE
```
Comment (jwhui) [inline src/core/thread/indirect_sender.cpp:85]: Combine with above `#if OPENTHREAD_FTD`?

File: src/core/thread/indirect_sender.cpp [hunk 16]
```diff
@@ -163,7 +163,7 @@ void IndirectSender::ClearAllMessagesForSleepyChild(Child &aChild)
     mSourceMatchController.ResetMessageCount(aChild);
 
     mDataPollHandler.RequestFrameChange(DataPollHandler::kPurgeFrame, aChild);
-#if !OPENTHREAD_MTD && OPENTHREAD_CONFIG_MAC_CSL_TRANSMITTER_ENABLE
+#if OPENTHREAD_FTD && OPENTHREAD_CONFIG_MAC_CSL_TRANSMITTER_ENABLE
```
Comment (jwhui) [inline src/core/thread/indirect_sender.cpp:166]: Combine with above `#if OPENTHREAD_FTD`?

File: src/core/thread/indirect_sender.cpp [hunk 17]
```diff
@@ -208,7 +208,7 @@ void IndirectSender::HandleChildModeChange(Child &aChild, Mle::DeviceMode aOldMo
         mSourceMatchController.ResetMessageCount(aChild);
 
         mDataPollHandler.RequestFrameChange(DataPollHandler::kPurgeFrame, aChild);
-#if !OPENTHREAD_MTD && OPENTHREAD_CONFIG_MAC_CSL_TRANSMITTER_ENABLE
+#if OPENTHREAD_FTD && OPENTHREAD_CONFIG_MAC_CSL_TRANSMITTER_ENABLE
```
Comment (jwhui) [inline src/core/thread/indirect_sender.cpp:211]: Combine with above `#if OPENTHREAD_FTD`?

File: src/core/thread/indirect_sender.cpp [hunk 18]
```diff
@@ -463,7 +463,7 @@ void IndirectSender::HandleSentFrameToChild(const Mac::TxFrame &aFrame,
     {
         aChild.SetIndirectFragmentOffset(nextOffset);
         mDataPollHandler.HandleNewFrame(aChild);
-#if !OPENTHREAD_MTD && OPENTHREAD_CONFIG_MAC_CSL_TRANSMITTER_ENABLE
+#if OPENTHREAD_FTD && OPENTHREAD_CONFIG_MAC_CSL_TRANSMITTER_ENABLE
```
Comment (jwhui) [inline src/core/thread/indirect_sender.cpp:466]: Combine with above `#if OPENTHREAD_FTD`?

File: src/core/thread/indirect_sender.cpp [hunk 19]
```diff
@@ -321,7 +321,7 @@ void IndirectSender::UpdateIndirectMessage(Child &aChild)
     aChild.SetIndirectFragmentOffset(0);
     aChild.SetIndirectTxSuccess(true);
 
-#if !OPENTHREAD_MTD && OPENTHREAD_CONFIG_MAC_CSL_TRANSMITTER_ENABLE
+#if OPENTHREAD_FTD && OPENTHREAD_CONFIG_MAC_CSL_TRANSMITTER_ENABLE
```
Comment (jwhui) [inline src/core/thread/indirect_sender.cpp:324]: Combine with above `#if OPENTHREAD_FTD`?

File: src/core/thread/indirect_sender.cpp [hunk 20]
```diff
@@ -295,7 +295,7 @@ void IndirectSender::RequestMessageUpdate(Child &aChild)
 
     aChild.SetWaitingForMessageUpdate(true);
     mDataPollHandler.RequestFrameChange(DataPollHandler::kReplaceFrame, aChild);
-#if !OPENTHREAD_MTD && OPENTHREAD_CONFIG_MAC_CSL_TRANSMITTER_ENABLE
+#if OPENTHREAD_FTD && OPENTHREAD_CONFIG_MAC_CSL_TRANSMITTER_ENABLE
```
Comment (jwhui) [inline src/core/thread/indirect_sender.cpp:298]: Combine with above `#if OPENTHREAD_FTD`?

File: src/core/thread/indirect_sender.cpp [hunk 21]
```diff
@@ -263,7 +263,7 @@ void IndirectSender::RequestMessageUpdate(Child &aChild)
 
         aChild.SetWaitingForMessageUpdate(true);
         mDataPollHandler.RequestFrameChange(DataPollHandler::kPurgeFrame, aChild);
-#if !OPENTHREAD_MTD && OPENTHREAD_CONFIG_MAC_CSL_TRANSMITTER_ENABLE
+#if OPENTHREAD_FTD && OPENTHREAD_CONFIG_MAC_CSL_TRANSMITTER_ENABLE
```
Comment (jwhui) [inline src/core/thread/indirect_sender.cpp:266]: Combine with above `#if OPENTHREAD_FTD`?

File: src/core/thread/topology.cpp [hunk 22]
```diff
@@ -234,7 +234,7 @@ void Child::Info::SetFrom(const Child &aChild)
     mFullThreadDevice   = aChild.IsFullThreadDevice();
     mFullNetworkData    = aChild.IsFullNetworkData();
     mIsStateRestoring   = aChild.IsStateRestoring();
-#if !OPENTHREAD_MTD && OPENTHREAD_CONFIG_MAC_CSL_TRANSMITTER_ENABLE
+#if OPENTHREAD_FTD && OPENTHREAD_CONFIG_MAC_CSL_TRANSMITTER_ENABLE
```
Comment (jwhui) [inline src/core/thread/topology.cpp:237]: Combine with above `#if OPENTHREAD_FTD`?

Comment (Irving-cl) [reply src/core/thread/topology.cpp:237]: It looks like there isn't any other `#if OPENTHREAD_FTD` above for this #if define.

**Review by jwhui — APPROVED** [review]

**Pull request merged**